src/base/base.c

```diff
@@ -1067,6 +1101,83 @@ static zend_observer_fcall_handlers xdebug_observer_init(zend_execute_data *exec
 	return (zend_observer_fcall_handlers){xdebug_execute_begin, xdebug_execute_end};
 }
 #endif
+
+static void xdebug_enable_debugger_handlers()
+{
+#if PHP_VERSION_ID >= 80100
+	zend_execute_ex = xdebug_execute_ex;
+#endif
+	
+	xdebug_enable_statement_extension_handler();
+}
+
+static void xdebug_disable_debugger_handlers()
```

Comment:
This function is called when we want to disable the debugger. We remove the normal function handler if using PHP 8.1+ as all the functionality we need in this mode can be handled by the handlers called by the Observer API. If it is PHP 8.0 we need to leave it in place.